libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -626,8 +626,10 @@ void AP_GPS::update_instance(uint8_t instance)
             state[instance].hdop = GPS_UNKNOWN_DOP;
             state[instance].vdop = GPS_UNKNOWN_DOP;
             timing[instance].last_message_time_ms = tnow;
+            timing[instance].delta_time_ms = UINT16_MAX; // we are going to lose lock on what this is until after we have parsed 2 messages
         }
     } else {
+        timing[instance].delta_time_ms = MIN(tnow - timing[instance].last_message_time_ms, UINT16_MAX); // clamp the initial reading
```

Comment:
I'd rather have the very clear distinction between no data and data. Although you have a point that we could clamp to the timeout value and it would still be sane. Will update.

Reply:
I think the way it is now is probably the best, but the change you made wasn't actually what I was proposing. I've seen now that I was wrong in my thinking though.

Let me explain: I thought that `last_message_time_ms` was zero before passing here a first time, so my thought was that, when there wasn't a driver, the delay value would be UINT16_MAX and, when there was, the first value would be 4000 (because `tnow - last_message_time` would be bigger in the first time).
I see now that `last_message_time` is actually set to *now* when we create the driver. So my question now is, why do we need the clamp? I don't see when `tnow - last_message_time` would ever be great than 4000.

Reply:
If it takes more then 4 seconds to detect the GPS after it was lost (which is possible) it could come out to be a much larger value.

Reply:
If it takes more than 4 seconds then the driver will be deleted. Which is why I can't see any real code path that makes the clamp needed. The only possible code path would be if after the driver creation we wouldn't call the GPS update method for more than 4 seconds - but I think that would be a problem, right?